### ISSUE TITLE
x/build,cmd/link: darwin/amd64 builders unexpectedly skipping `cmd/link.TestBuildForTvOS`

### ISSUE THREAD

**Issue by bcmills** [golang/go#58231]
In #58221 we observed consistent test failures in `cmd/link.TestBuildForTvOS` on `darwin-amd64-11_0`, but not on the other `darwin/amd64` builders. The root cause of the failure is not OS-dependent, which suggests that the test is being skipped on all of the other `darwin/amd64` builders. That is unexpected.

We should probably:
- adjust those builders so that they can run the test (attn @golang/release), and
- adjust the test so that it fails instead of skipping if `GO_BUILDER_NAME` is non-empty (attn @golang/compiler).

(The discussion on #58221 covers more detail about the skips.)

**Comment by cherrymui** [comment]
Yeah, the test requires tvOS SDK installed on the machine. Not sure they are...

**Comment by heschi** [comment]
cc @prattmic for the next time we build macOS images?

**Comment by heschi** [comment]
@golang/ios How much do we care about this test?

**Comment by changkun** [comment]
cc @eliasnaur

**Comment by prattmic** [comment]
The different builders have slightly different versions of XCode installed (generally the latest that happens to be supported by that OS version). It wouldn't surprise me if one of those XCode installations isn't quite right. Exact installation steps seemed to vary slightly from version to version, so it wouldn't surprise me if something went wrong along the way.

**Comment by dmitshur** [comment]
Xcode completes its installation on its first launch, showing a prompt that looks like this (in a recent version):

<img width="612" alt="image" src="https://user-images.githubusercontent.com/1924134/217581885-9e7572ff-eed3-4aa1-a152-fb3b3696597a.png">

So it'd be a matter of not forgetting to check the box next to tvOS and pressing continue next time.

**Comment by prattmic** [comment]
I may have done the installation incorrectly, but FWIW I never saw that window on any of the Xcode installations I did for Darwin builders. IIRC, the [installation process](https://cs.opensource.google/go/x/build/+/master:env/darwin/setup-notes.md;l=94-104) is completely GUI-free other than accepting the license.
